src/Arguments.ts

```diff
@@ -37,18 +37,15 @@ export class Arg {
     static any<T extends 'boolean'>(type: T): Argument<boolean> & boolean
     static any<T extends 'array'>(type: T): Argument<any[]> & any[]
     static any<T extends 'function'>(type: T): Argument<Function> & Function
-    static any<T extends 'string'|'number'|'boolean'|'symbol'|'undefined'|'object'|'function'|'array'>(type: T): Argument<any> & any
+    static any<T extends 'string' | 'number' | 'boolean' | 'symbol' | 'undefined' | 'object' | 'function' | 'array'>(type: T): Argument<any> & any
     static any(type?: string): Argument<any> & any {
         const description = !type ? '{any arg}' : '{type ' + type + '}';
         return new Argument<any>(description, x => {
-            if(!type)
+            if (!type)
                 return true;
 
-            if(typeof x === 'undefined')
-                return true;

```

Comment:
I removed this as I see it as a bug. If we explicitly pass in the type on `Arg.any(type: T)`, we should always compare the type of x to it. Also if we expect x to be undefined both `Arg.any()` and `Arg.any('undefined')` will be true.

Reply:
But maybe it's there for a reason I am overseeing. So that's why I want to ask if it's okay to remove.

Reply:
Perfect! Well, as long as there is no test describing why the change is there, I'd say we should remove it :+1:

Nicely done.

Reply:
Cool! I will also add an Arguments spec, so we can have it tested 😄